File: 48-rotate-image/48-rotate-image.cpp
class Solution {
public:
    void rotate(vector<vector<int>>& matrix) {
        //swap across diagonal
        for(int i = 0; i < matrix.size(); i++){
            for(int j = 0; j < matrix[0].size(); j++){
                if(j > i){
                    swap(matrix[i][j], matrix[j][i]);
                }
            }
        }
        
        //mirror image
        int mid = matrix[0].size()/2;
        
        for(int c = 0; c < mid; c++){
            for(int r = 0; r < matrix.size(); r++){
                swap(matrix[r][c], matrix[r][matrix.size() - c - 1]);
            }
        }
    }
};

File: 48-rotate-image/48-rotate-image.py
class Solution:
    def rotate(self, matrix: List[List[int]]) -> None:
        
        #Transpose of the matrix
        for i in range(len(matrix)):
            for j in range(i+1):
                matrix[i][j], matrix[j][i] = matrix[j][i], matrix[i][j]
                
        #reverse each row
        for array in matrix:
            array.reverse()
            
        